src/include/duckdb/execution/merge_sort_tree.hpp

```diff
@@ -44,6 +46,30 @@ namespace duckdb {
 // Implementation of a generic merge-sort-tree
 // Rewrite of the original, which was in C++17 and targeted for research,
 // instead of deployment.
+template <typename T>
```

Comment:
Since we adopted this code, would it make sense to separate some of the functionality into source files? I am thinking about modularity and also readability.

Reply:
Yeah that's been on my radar. Right now there is a bit of code reuse, so maybe add this to the cleanup issue?